neqo-server/src/main.rs

```diff
@@ -660,14 +653,14 @@ impl ServersRunner {
     /// Init Poll for all hosts. Create sockets, and a map of the
     /// socketaddrs to instances of the HttpServer handling that addr.
     fn init(&mut self) -> Result<(), io::Error> {
-        self.hosts = self.args.listen_addresses();
-        if self.hosts.is_empty() {
+        let hosts = self.args.listen_addresses();
+        if hosts.is_empty() {
             eprintln!("No valid hosts defined");
             return Err(io::Error::new(io::ErrorKind::InvalidInput, "No hosts"));
         }
 
-        for (i, host) in self.hosts.iter().enumerate() {
-            let socket = match UdpSocket::bind(host) {
+        for host in hosts.into_iter() {
```

Comment:
```suggestion
        for host in hosts {
```

I believe that for ... in already calls `IntoIterator::into_iter()`.

Reply:
:+1: simplified further using `Iterator` chain 2821ed4c367430c4eaca96fe189950bff2d0ddaa. Let me know if you find this easier to read.